include/zephyr/drivers/charger.h

```diff
@@ -173,6 +197,16 @@ enum charger_health {
 	CHARGER_HEALTH_NO_BATTERY,
 };
 
+/**
+ * @brief The input current thresholds for the charger to notify the system
+ */
+struct charger_input_current_notifier {
+	/** The input current threshold to be exceeded */
+	uint32_t current_ua;
```

Comment:
Do we want to name these like `max_threshold_current_ua` so it's easier for us to add something like a `min_threshold` or is that unlikely to be necessary?

Reply:
@aaronemassey,

I'd say that feels unlikely. What were notifying the system of here is "hey, we've been pulling X amps for Y µs, throttle your clock speed so we don't crash our adaptor" so I don't see a minimum threshold use-case.

I did consider what if we need `voltage_uv` but those thresholds are not time-dependent (at least in x86-PROCHOT land) and that does make sense.

It might be smart to rename `charger_input_current_notifier` to ` charger_current_event_notifier` as there are notifications related to battery discharge on some systems.

How do you feel about adding a member of the struct to indicate severity level instead of there being a different property (CHARGER_PROP_INPUT_CURRENT_WARNING, CHARGER_PROP_INPUT_CURRENT_CRITICAL, etc.) for each threshold?

Reply:
> I think that is much better because we might be able to avoid polling the chip via the driver again.
Done, let me know what you think.

Reply:
Seems reasonable. But shouldn't we have some default API described severity levels? Like the one you mentioned like `CRITICAL`?

Reply:
I've seen this feature implemented on chargers with two to three levels. I can add something like:
``` c
enum charger_notification_severity {
    CHARGER_SEVERITY_PEAK = 0;
    CHARGER_SEVERITY_CRITICAL;
    CHARGER_SEVERITY_WARNING;
};
```

Reply:
LGTM

Reply:
done!